pkg/provider/azure.go

```diff
@@ -573,14 +573,18 @@ func (az *Cloud) InitializeCloudFromConfig(config *Config, fromSecret, callFromC
 		az.routeUpdater = newDelayedRouteUpdater(az, routeUpdateInterval)
 		go az.routeUpdater.run()
 
-		// wait for the success first time of syncing zones
-		err = az.syncRegionZonesMap()
-		if err != nil {
-			klog.Errorf("InitializeCloudFromConfig: failed to sync regional zones map for the first time: %s", err.Error())
-			return err
-		}
+		// Azure Stack does not support zone at the moment
+		// https://docs.microsoft.com/en-us/azure-stack/user/azure-stack-network-differences?view=azs-2102
+		if !az.isStackCloud() {
```

Comment:
Could we add another configure for this feature? Some deployments on Azure are using AzureStack config to customize ARM endpoints, hence this check would break those deployments

Reply:
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/provider/azure.go#L175 -
```
	// DisableAzureStackCloud disables AzureStackCloud support. It should be used
	// when setting AzureAuthConfig.Cloud with "AZURESTACKCLOUD" to customize ARM endpoints
	// while the cluster is not running on AzureStack.
	DisableAzureStackCloud bool `json:"disableAzureStackCloud,omitempty" yaml:"disableAzureStackCloud,omitempty"
```

`isStackCloud` also check `disableAzureStackCloud` config parameter. This behaviour might be disabled using this config parameter, as described in comment.

Reply:
thanks for adding this. LGTM then.